src/latexify/codegen/expression_rules.py

```diff
@@ -0,0 +1,266 @@
+        "}",
+        operand_left=BinOperandRule(force=True),
+        operand_right=BinOperandRule(wrap=False),
+    ),
+    ast.Mult: BinOpRule("", " ", ""),
+    ast.MatMult: BinOpRule("", " ", ""),
+    ast.Div: BinOpRule(
+        r"\frac{",
+        "}{",
+        "}",
+        operand_left=BinOperandRule(wrap=False),
+        operand_right=BinOperandRule(wrap=False),
+    ),
+    ast.FloorDiv: BinOpRule(
+        r"\left\lfloor\frac{",
+        "}{",
+        r"}\right\rfloor",
+        operand_left=BinOperandRule(wrap=False),
+        operand_right=BinOperandRule(wrap=False),
+        is_wrapped=True,
+    ),
+    ast.Mod: BinOpRule(
+        "", r" \mathbin{\%} ", "", operand_right=BinOperandRule(force=True)
+    ),
+    ast.Add: BinOpRule("", " + ", ""),
+    ast.Sub: BinOpRule("", " - ", "", operand_right=BinOperandRule(force=True)),
+    ast.LShift: BinOpRule("", r" \ll ", "", operand_right=BinOperandRule(force=True)),
+    ast.RShift: BinOpRule("", r" \gg ", "", operand_right=BinOperandRule(force=True)),
+    ast.BitAnd: BinOpRule("", r" \mathbin{\&} ", ""),
+    ast.BitXor: BinOpRule("", r" \oplus ", ""),
+    ast.BitOr: BinOpRule("", r" \mathbin{|} ", ""),
+}
+
+# Typeset for BinOp of sets.
+SET_BIN_OP_RULES: dict[type[ast.operator], BinOpRule] = {
+    **BIN_OP_RULES,
+    ast.Sub: BinOpRule(
+        "", r" \setminus ", "", operand_right=BinOperandRule(force=True)
+    ),
+    ast.BitAnd: BinOpRule("", r" \cap ", ""),
+    ast.BitXor: BinOpRule("", r" \mathbin{\triangle} ", ""),
+    ast.BitOr: BinOpRule("", r" \cup ", ""),
+}
+
+UNARY_OPS: dict[type[ast.unaryop], str] = {
+    ast.Invert: r"\mathord{\sim} ",
+    ast.UAdd: "+",  # Explicitly adds the $+$ operator.
+    ast.USub: "-",
+    ast.Not: r"\lnot ",
+}
+
+COMPARE_OPS: dict[type[ast.cmpop], str] = {
+    ast.Eq: "=",
+    ast.Gt: ">",
+    ast.GtE: r"\ge",
+    ast.In: r"\in",
+    ast.Is: r"\equiv",
+    ast.IsNot: r"\not\equiv",
+    ast.Lt: "<",
+    ast.LtE: r"\le",
+    ast.NotEq: r"\ne",
+    ast.NotIn: r"\notin",
+}
+
+# Typeset for Compare of sets.
+SET_COMPARE_OPS: dict[type[ast.cmpop], str] = {
+    **COMPARE_OPS,
+    ast.Gt: r"\supset",
+    ast.GtE: r"\supseteq",
+    ast.Lt: r"\subset",
+    ast.LtE: r"\subseteq",
+}
+
+BOOL_OPS: dict[type[ast.boolop], str] = {
+    ast.And: r"\land",
+    ast.Or: r"\lor",
+}
+
+
+@dataclasses.dataclass(frozen=True)
+class FunctionRule:
+    """Codegen rules for functions.
+
+    Attributes:
+        left: LaTeX expression concatenated to the left-hand side of the arguments.
+        right: LaTeX expression concatenated to the right-hand side of the arguments.
+        is_unary: Whether the function is treated as a unary operator or not.
+        is_wrapped: Whether the resulting syntax is wrapped by brackets or not.
+    """
+
+    left: str
+    right: str = ""
+    is_unary: bool = False
+    is_wrapped: bool = False
+
+
+# name => left_syntax, right_syntax, is_wrapped
+BUILTIN_FUNCS: dict[str, FunctionRule] = {
```

Comment:
Assuming this constant doesn't have a `_` prefix since it's imported from another file. In that case, shouldn't `_MATH_SYMBOLS` in `math_symbols.py` also have it's `_` prefix removed?

Reply:
moved `_MATH_SYMBOLS` to `expression_rules.MATH_SYMBOLS`